frontend/simple/views/CreateGroup.vue

```diff
@@ -3,188 +3,78 @@
-                    <span class="select">
-                      <select>
-                        <option>USD</option>
-                        <option>BTC</option>
-                        <option>EUR</option>
-                      </select>
-                    </span>
-                  </p>
-                  <p class="control">
-                    <input class="input" type="text" name="incomeProvided"
-                      placeholder="Amount"
-                      data-vv-as="Income Provided"
-                      v-validate data-vv-rules="required|decimal:2"
-                      v-model="incomeProvided"
-                    />
-                  </p>
-                </div>
-                <i18n v-if="errors.has('incomeProvided')" data-control="incomeProvided" class="help is-danger">
-                  The Income Provided field must be a numeric currency amount and may contain 2 decimal points.
-                </i18n>
-              </div>
-              <div class="box" >
-                <p class="title is-5"><i18n>How transparent should your group be about who contributes?</i18n></p>
-                <p class="select">
-                  <select
-                    v-validate
-                    data-vv-rules="required"
-                    data-vv-as="Contribution Privacy"
-                    name="contributionPrivacy"
-                    v-model="contributionPrivacy"
-                  >
-                    <option value="">Select an option</option>
-                    <option value="Very Private">Very Private</option>
-                  </select>
-                </p>
-                <i18n v-if="errors.has('contributionPrivacy')" data-control="contributionPrivacy" class="help is-danger">
-                  {{ errors.first('contributionPrivacy') }}
-                </i18n>
-              </div>
-              <div class="has-text-centered button-box">
-                <div id="errorMsg" v-if="errorMsg" class="help is-danger">{{ errorMsg }}</div>
-                <button
-                  class="button is-success is-large"
-                  type="submit"
-                >
-                  <i18n>Create Group</i18n>
-                </button>
-              </div>
-            </div>
+      <div class="column is-half is-offset-one-quarter" >
+        <transition name="fade" mode="out-in">
+          <router-view :group="form" :validity="validity" @input="(payload) => updateGroupData(payload)">
+          </router-view>
+        </transition>
+
+        <div class="field step-controls">
+          <p class="control" v-if="currentStep + 1 < config.steps.length">
+            <button
+              class="button is-success is-large"
+              @click.prevent="next"
+              id="nextBtn"
+            >
+              <i18n>Next</i18n>
+              <span class="icon">
+                <i class="fa fa-arrow-right"></i>
+              </span>
+            </button>
```

Comment:
Ideally, using vee-validate, the next button would highlight and select the invalid / unfilled form element in a red outline, and refuse to switch to the next view.

It would also mean you wouldn't have to tell the user `L('We still need some info from you, please go back and fill missing fields')`.

Reply:
you are right, right now we don't indicate required or erroneous fields anywhere, that should definitely be fixed.

but on the 'refuse to switch to the next view' part: I intentionally went with a more allowing user flow in this case, where a user could look through all the steps at first to familiarize with the process, and then go back to any step to fill it. I see no gain in blocking and stressing the user ahead in the process.

until now, all the questions were visible at the same time, so the user knew what's ahead. also, there, progressing without filling something would have broken group creation. there that was the ideal solution. Now I see room for discussion.

@willstanley what is your opinion on this from the ux point of view?

Reply:
IMO, the way I've seen other apps and websites do it is: when a required field is not filled in, it is highlighted in red and oftentimes an effect is applied (e.g. focusing the cursor on the field and making the field itself bounce, for example).

Personally, I don't think users will have much stress from that, as they're used to it and expect required fields to be filled in. Also curious what @willstanley thinks.

Reply:
If it’s a required field, I would prevent the user from moving to the next screen until it’s been filled in.

I think going back through the flow to find the field you’ve missed would cause more hassle for the user.